lms/djangoapps/courseware/views.py

```diff
@@ -609,14 +611,30 @@ def jump_to(request, course_id, location):
     # choose the appropriate view (and provide the necessary args) based on the
     # args provided by the redirect.
     # Rely on index to do all error handling and access control.
+
     if chapter is None:
-        return redirect('courseware', course_id=course_key.to_deprecated_string())
+        redirect_url = reverse('courseware', args=(unicode(course_key), ))
     elif section is None:
-        return redirect('courseware_chapter', course_id=course_key.to_deprecated_string(), chapter=chapter)
+        redirect_url = reverse('courseware_chapter', args=(unicode(course_key), chapter))
     elif position is None:
-        return redirect('courseware_section', course_id=course_key.to_deprecated_string(), chapter=chapter, section=section)
+        redirect_url = reverse(
+            'courseware_section',
+            args=(unicode(course_key), chapter, section)
+        )
     else:
-        return redirect('courseware_position', course_id=course_key.to_deprecated_string(), chapter=chapter, section=section, position=position)
+        # Here we use the navigation_index from the position returned from
+        # path_to_location - we can only navigate to the topmost vertical at the
+        # moment
+
+        redirect_url = reverse(
+            'courseware_position',
+            args=(unicode(course_key), chapter, section, position)
+        )
+
+    if final_target_id:
+        redirect_url += "?activate_block_id={final_target_id}".format(final_target_id=final_target_id)
```

Comment:
I think you need to port the upstream fix here, to remove `final_target_id` and to url-encode this. Though when I tested on old mongo courses in Apros and the LMS it worked fine, since they don't have `+` in their URLs.